src/formatter-settings/assets/features/formatterSettings/components/Setting.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import checkIcon from "@iconify-icons/codicon/check";
+import chevronDownIcon from "@iconify-icons/codicon/chevron-down";
+import { Icon } from "@iconify/react";
+import React, { Dispatch } from "react";
+import { Dropdown, Form } from "react-bootstrap";
+import { useDispatch } from "react-redux";
+import { Catagory, JavaFormatterSetting, ValueKind } from "../../../../types";
+import { changeSetting } from "../formatterSettingViewSlice";
+
+export interface SettingProps {
+  setting: JavaFormatterSetting[];
+  catagory: Catagory;
+}
+
+const Setting = (prop: SettingProps): JSX.Element => {
+
+  const dispatch: Dispatch<any> = useDispatch();
+
+  const handleChangeCheckbox = (e: any) => {
+    const id = e.target.id;
+    const value = (e.target.checked === true) ? "true" : "false";
+    dispatch(changeSetting({id: id, value: value}));
+  };
+
+  const handleChangeInput = (e: any) => {
+    const id = e.target.id;
+    let value = e.target.value;
+    if (!value || value === "") {
+      value = "0";
+    }
+    dispatch(changeSetting({id: id, value: value}));
+  };
+
+  const handleSelect = (setting: JavaFormatterSetting, entry: string) => {
+    dispatch(changeSetting({id: setting.id, value: entry}));
+  };
+
+  const generateSetting = (setting: JavaFormatterSetting) => {
+    if (!setting.name || !setting.id || !setting.value) {
+      return;
+    }
+    const candidates = [];
+    switch (setting.valueKind as ValueKind) {
+      case ValueKind.Boolean:
```

Comment:
If you quickly click the checkbox multiple times, the label will also be selected. Can we avoid this?

Reply:
Sure. Will set `user-select` to `none` in `.form-check-label`.